parser_library/test/common_testing.h

```diff
@@ -216,6 +227,15 @@ inline bool contains_message_text(const std::vector<Msg>& d, const C& c)
     return contains_message_properties(d, c, &Msg::message);
 }
 
+template<typename Msg, typename C = std::initializer_list<std::string>>
+inline bool matches_partial_message_text(const std::vector<Msg>& d, const C& c)
+{
+    return matches_message_properties(
+        d, c, &Msg::message, [](const decltype(Msg::message)& a, const decltype(Msg::message)& b) -> bool {
+            return a.find(b) != decltype(Msg::message)::npos;
+        });
```

Comment:
does this work?
```suggestion
        d, c, &Msg::message, [](const auto& a, const auto& b) {
            return a.find(b) != a.npos;
        });
```
or
```suggestion
        d, c, &Msg::message, []<typename T>(const T& a, const auto& b) {
            return a.find(b) != T::npos;
        });
```
or I think that what we are looking for is obviously "stringy" so 

```suggestion
        d, c, &Msg::message, [](std::string_view a, std::string_view b) {
            return a.find(b) != std::string_view::npos;
        });
```

Reply:
I opted for `std::string_view` due to its expressiveness